File: cc-go/src/middlewares/middleware.go
package middlewares

import (
	"github.com/gofiber/fiber/v2"
)

func PageNotFound(c *fiber.Ctx) error {
	return c.Next()
}

func ErrorHandler(c *fiber.Ctx) error {
	statusCode := c.Status(200)

	if statusCode == 200 {
		statusCode = 500
	} else {
		statusCode = statusCode
	}
	return c.Status(statusCode).JSON(fiber.Map{
		"error": c.Error(),
	})
}

File: cc-node/src/middlewares/middleware.ts
import { Request, Response, NextFunction } from "express";

export function pageNotFound(req: Request, res: Response, next: NextFunction) {
  const error = new Error(`${req.originalUrl} page not found!`);
  res.status(404);
  next(error.message);
}

export function errorHandler(
  error: Error,
  req: Request,
  res: Response,
  next: NextFunction
) {
  const statusCode = res.statusCode === 200 ? 500 : res.statusCode;
  res.status(statusCode);
  res.send({
    message: error.message,
    stack: process.env.NODE_ENV === "development" ? error.stack : "",
  });
}
